static-build/posts/2021/09/cors/index.md

```diff
@@ -0,0 +1,354 @@
+
+Netscape realised that this presented a security issue, so they decided that cross-frame scripting would only be allowed if both pages had the same _origin_.
+
+<style>
+.origin-example {
+  display: grid;
+  grid-template-columns: max-content max-content;
+}
+.origin {
+  grid-area: 1 / 1;
+  color: #ff824a;
+}
+.path {
+  grid-area: 1 / 2;
+}
+.origin-label {
+  grid-area: 2 / 1;
+  text-align: center;
+  display: grid;
+  grid-template-columns: 1fr max-content 1fr;
+  align-items: top;
+  gap: 0.3em;
+  color: #ff824a;
+}
+.origin-label::before,
+.origin-label::after {
+  content: '';
+  border: 1px solid currentcolor;
+  height: 50%;
+}
+.origin-label::before {
+  border-width: 0 0 1px 1px;
+}
+.origin-label::after {
+  border-width: 0 1px 1px 0;
+}
+</style>
+
+<div class="code-example"><pre class="language-xml"><code><div class="origin-example"><div aria-labelledby="origin-label" class="origin">https://jakearchibald.com:443</div><div class="path">/2021/blah/?foo#bar</div><div id="origin-label" class="origin-label">The origin</div></div></code></pre></div>
+
+The idea was that sites on the same origin are more likely to have the same owner. That wasn't completely true, since a lot of sites divided content by URLs such as `http://example.com/~jakearchibald/`, but the line had to be drawn somewhere.
+
+From that point, features that granted deep visibility into a resource were limited to same-origin. This included `new ActiveXObject('Microsoft.XMLHTTP')` which first appeared in IE5 in 1999, and later became the web standard `XMLHttpRequest`.
+
+## Origins vs sites
+
+Some web features don't deal with origins, they deal with 'sites'. For instance, `https://help.yourbank.com` and `https://profile.yourbank.com` are different origins, but they're the same site. Cookies are the most common feature that operate at a site level, as you can create cookies that are sent to all subdomains of `yourbank.com`.
+
+But how does the browser know that `https://help.yourbank.com` and `https://profile.yourbank.com` are part of the same site, but `https://yourbank.co.uk` and `https://jakearchibald.co.uk` are different sites? I mean… they all have three parts separated by dots.
+
+Well, the answer was a bunch of heuristics in each browser, but in 2007 Mozilla swapped their heuristics for a list. That list is now maintained as a separate community project known as the [public suffix list](https://publicsuffix.org/), and it's used by all browsers and many other projects.
+
+If someone says they understand the security implications of URLs without UI hints, be sure to check they can recite all 9000+ entries of the public suffix list from memory.
+
+# Opening things up again
+
+Ok, so we've got these APIs like `<img>` that can access resources from other origins, but visibility into the response is limited (but not limited enough in hindsight), and we've got these more powerful APIs like cross-frame scripting and `XMLHttpRequest` which only work same-origin.
```

Comment:
Link "not limited enough in hindsight" to something?

Reply:
It's referring to the details earlier in the article (like leaking login state, and the Yahoo attack)

Reply:
Oh, OK. I thought you were referring to Spectre/Meltdown.

Reply:
I should probably at least mention those.